### ISSUE TITLE
Add a LICENSE

### ISSUE THREAD

**Issue by Ryuno-Ki** [dimka665/awesome-slugify#24]
In order for others to contribute or use this work, it would be handy to have a LICENSE file declaring, under which conditions it can be included.

You can, for example, use [Choose A License](http://choosealicense.com/) service to guide you with your choice ;-)

**Comment by jonafato** [comment]
The license listed on the [cheeseshop](https://pypi.python.org/pypi/awesome-slugify/1.6.5) is GPLv3. While that's probably sufficient for those looking, it'd be nice to see one in the GitHub repository as well.

**Comment by dimka665** [comment]
Hi all.
I prefer to distribute my libraries under MIT or BSD license. But awesome-slufigy requires `unidecode` library that has GPL license. Can I use MIT license in this case?

**Comment by jonafato** [comment]
(Obligatory preface: I am not a lawyer, and none of my comments based on reading some FAQs on the internet constitute legal advice.) This is a pretty complex question. Per [this FAQ entry](https://www.gnu.org/licenses/gpl-faq.html#LinkingWithGPL):
> **You have a GPL'ed program that I'd like to link with my code to build a proprietary program. Does the fact that I link with your program mean I have to GPL my program?**
> Not exactly. It means you must release your program under a license compatible with the GPL (more precisely, compatible with one or more GPL versions accepted by all the rest of the code in the combination that you link). The combination itself is then available under those GPL versions.

My naive interpretation of that answer is that the **combined** work (i.e. a binary or source distribution of the library **and** its dependencies) must be licensed under the GPL, but the library itself need only be licensed under a GPL compatible license ([see also](https://www.gnu.org/licenses/gpl-faq.html#IfLibraryIsGPL)). Because you're depending on a library but not actually distributing a combined work, you likely only need to use one of the compatible licenses rather than the GPLv3 itself. Among those compatible licenses are (I'm assuming these cover the options being considered under "MIT or BSD" above):

- [3-clause BSD](https://www.gnu.org/licenses/license-list.en.html#ModifiedBSD)
- [2-clause BSD](https://www.gnu.org/licenses/license-list.en.html#FreeBSD)
- [Expat](https://www.gnu.org/licenses/license-list.en.html#Expat) ("It is sometimes ambiguously referred to as the MIT License.")
- [X11](https://www.gnu.org/licenses/license-list.en.html#X11License) ("This license is sometimes called the MIT license")

Based on those links, it seems like as long as you pick a GPL compatible license for the library, it should be compliant with the terms of the dependency's license.

**TL;DR**: you're probably fine choosing one of those licenses for this library.

**Comment by Ryuno-Ki** [comment]
Yepp, I saw GPL v3 on PyPI as well, but missed it here.

**Comment by jeffwidman** [comment]
I'd much prefer you release this under MIT/BSD, as GPL v3 isn't allowed on a project I'm working on. Is that possible?

**Comment by Ryuno-Ki** [comment]
@jonafato Be aware, that you can release the source code under multiple licenses as well as grant a derivating license to certain people (preferably in written form).

Disclaimer: I'm not a lawyer either.

**Comment by jonafato** [comment]
Any update here? I'd love to see a license file added to this project.

**Comment by ergo** [comment]
Another not laywer - but I think adding a GPL library to existing project instantly makes it GPL too so @jonafato is right.

**Comment by ergo** [comment]
https://pypi.python.org/pypi/text-unidecode/1.0 - maybe you could switch to this library and then you could do MIT/Apache2.0/BSD?

**Comment by jonafato** [comment]
@ergo I was suggesting that this library did not need to be GPL-licensed because no GPL code is directly included. Depending on third party code that is installed by a package manager should not affect the GPL licensing requirements the same way as directly including the same code for redistribution.

**Comment by Jenselme** [comment]
Any progress on this? This library is used by [agate](https://github.com/wireservice/agate). I am in the process of packaging it (and thus awesome-slugify) for [fedora](http://fedoraproject.org/).

It is recommended for packages to have a license file prior to their inclusion in the distribution. Can you at least add the GPL text. I also think that for GPL software, headers are required in sources files too. From https://www.gnu.org/licenses/gpl-howto.en.html:

> Whichever license you plan to use, the process involves adding two elements to each source file of your program: a copyright notice (such as “Copyright 1999 Terry Jones”), and a statement of copying permission, saying that the program is distributed under the terms of the GNU General Public License (or the Lesser GPL).
